ci/travis/ci.sh

```diff
@@ -206,13 +220,42 @@ build_wheels() {
       suppress_output "${WORKSPACE_DIR}"/python/build-wheel-macos.sh
       ;;
     msys*)
-      if ! python -c "import ray" 2> /dev/null; then
-        # This installs e.g. the pickle5 dependencies, which are needed for correct wheel testing
-        install_ray
-      fi
       (
-        cd "${WORKSPACE_DIR}"/python
-        python setup.py --quiet bdist_wheel
+        local backup_conda="${CONDA_PREFIX}.bak" ray_uninstall_status=0
+        test ! -d "${backup_conda}"
+        pip uninstall -y ray || ray_uninstall_status=1
+        mv -n -T -- "${CONDA_PREFIX}" "${backup_conda}"  # Back up conda
+
+        local pyversion pyversions=()
+        for pyversion in 3.6 3.7 3.8; do
+          if [ "${pyversion}" = "${PYTHON-}" ]; then continue; fi  # we'll build ${PYTHON} last
+          pyversions+=("${pyversion}")
+        done
+
+        pyversions+=("${PYTHON-}")  # build this last so any subsequent steps use the right version
+        local local_dir="python/dist"
+        for pyversion in "${pyversions[@]}"; do
+          if [ -z "${pyversion}" ]; then continue; fi
+          "${ROOT_DIR}"/bazel-preclean.sh
```

Comment:
Can you describe why this is needed? We don't seem to be needing it for https://github.com/ray-project/ray/blob/master/python/build-wheel-manylinux1.sh

Reply:
I've already explained it [in the script](https://github.com/ray-project/ray/pull/8369/files/7e4969e07526399606fd3c55a3d0b5cd2b4c2512#diff-df651d5d46a7e5bb96618946d9d7323cR5), but the short version is that `git clean -f -f -x -d` without `bazel clean` doesn't cause Bazel to re-run any of our (hacky) genrules, because its outputs as Bazel sees them are never in the git worktree. I think that script is either broken (i.e. bundling the wrong version's wheels) or coincidentally re-triggering the rules in some way implicitly, but otherwise it has the same problem.

Reply:
Oh, maybe it doesn't run into this issue because its `PATH` changes in every run, so Bazel recompiles everything all over again (uncached). Whereas if we do this we don't need to recompile things that don't need to be recompiled.